doc/specs/Configuration object for profiles.md

```diff
@@ -0,0 +1,94 @@
+---
+author: <Pankaj> <Bhojwani> <pabhojwa@microsoft.com>
+created on: <2020-11-20>
+last updated: <2020-11-20>
+issue id: <#8345>
+---
+
+# Configuration object for profiles
+
+## Abstract
+
+This spec outlines how we can support 'configuration objects' in our profiles, which
+will allow us to render differently depending on the state of the control. For example, a
+control can be rendered differently if its focused as compared to when its unfocused. Another
+example is that an elevated state control can be rendered differently as compared to a
+non-elevated one.

```

Comment:
How does your design scale as we add new states? Consider the following profile:
```js
{
  "fontFace": "Cascadia Code",
  "state.unfocused" : {
    "fontFace": "Cascadia Mono"
  },
  "state.admin" : {
    "fontFace": "Consolas"
  }
}
```
What is the font face in the following scenarios:
| Unfocused? | Admin? | Font |
|--|--|--|
| ❌ | ❌ | Cascadia Code |
| ❌ | ✔ | Consolas |
| ✔ | ❌ | Cascadia Mono |
| ✔ | ✔ | ??? |